### PR TITLE
Add GraphQL topic to devdocs, closes #49

### LINKED ISSUE #49: GraphQL topic
_From @jcalcaben on April 5, 2018 15:37_

## This issue is a:
[ ] Bug
[X] New Topic Request
[ ] Topic Clarification Request
[ ] New Feature Request
[ ] Other
 

## Description:

Content needed for the **GraphQL** topic.

`src/technologies/basic-concepts/graphql/index.md`

## Expected Result:

**GraphQL** topic content exists and covers the following:

- [ ] What is GraphQL
- [ ] Benefits provided by GraphQL
- [ ] Why use GraphQL over REST
- [ ] Code samples provided for a schema, request, and response
 

## Additional Information:

- https://en.wikipedia.org/wiki/GraphQL
- https://graphql.org/
- https://medium.freecodecamp.org/so-whats-this-graphql-thing-i-keep-hearing-about-baf4d36c20cf

_Copied from original issue: magento-research/pwa-devdocs#10_

### PR DESCRIPTION
**Pull request by JStein92**
## This PR is a:

- [ ] New feature
- [ ] Enhancement/Optimization
- [ ] Refactor
- [ ] Bugfix
- [ ] Test for existing code
- [x] Documentation

## Summary

When this pull request is merged, it will add content to the **GraphQL** topic. 

src/technologies/basic-concepts/graphql/index.md

Contents of Topic:
- [x] What is GraphQL
- [x] Benefits Provided by GraphQL
- [x] Why Use GraphQL Over REST
- [x] Sample Queries
- [x] Learn More

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento-research/pwa-studio?pullRequest=228) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento-research/pwa-studio?pullRequest=228) before we can accept your contribution.<br/>

File: packages/pwa-devdocs/src/technologies/basic-concepts/graphql/index.md [hunk 1]
```diff
@@ -1,5 +1,347 @@
...
+
+## Why Use GraphQL Over REST
+
+While **GraphQL** and **REST** are both specifications for constructing and querying APIs, **GraphQL** has some significant advantages over REST.
+
+ - #### No Versioning
+
+  **REST** APIs will typically have multiple versions - v1, v2, etc. This is because updating endpoints in **REST** will often impact existing queries.
+
+  With **GraphQL**, there is no need for versioning, since new **types** and **fields** and can be added to the **schema** without impacting existing queries.
```
Comment (jcalcaben) [inline packages/pwa-devdocs/src/technologies/basic-concepts/graphql/index.md]: Remove the "and" after `**fields**`

**Comment by JStein92** [comment]
@jcalcaben Thank you for the feedback! I've pushed the requested changes.

**Comment by jcalcaben** [comment]
Thanks again for this contribution @JStein92 !

**Pull request merged**